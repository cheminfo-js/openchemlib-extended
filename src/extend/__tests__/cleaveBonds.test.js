'use strict';

const OCLE = require('../..');

describe('cleaveBonds test propane', () => {
  it('default', () => {
    var molecule = OCLE.Molecule.fromSmiles('CCC');
    let results = molecule.cleaveBonds({});
    expect(results).toMatchSnapshot();
  });

  it('charged', () => {
    var molecule = OCLE.Molecule.fromSmiles('CC[CH2+]');
    let results = molecule.cleaveBonds({});
    expect(results[0].fragments[1].mf).toMatch('C2H4(+)');
  });
});
