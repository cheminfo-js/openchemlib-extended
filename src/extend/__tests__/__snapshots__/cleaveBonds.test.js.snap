// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cleaveBonds test propane default 1`] = `
Array [
  Object {
    "atom1": 0,
    "atom2": 1,
    "fragments": Array [
      Object {
        "atomMap": Array [],
        "idCode": "eFBAYc@@",
        "mf": "CH3",
      },
      Object {
        "atomMap": Array [],
        "idCode": "eMBAYRZ@",
        "mf": "C2H5",
      },
    ],
    "hoses1": Array [
      Object {
        "f": "eFBAYcAqUA~ddxUP",
        "i": 0,
      },
      Object {
        "f": "eFBAYcAqUA~ddxUP",
        "i": 1,
      },
      Object {
        "f": "eFBAYcAqUA~ddxUP",
        "i": 2,
      },
    ],
    "hoses2": Array [
      Object {
        "f": "eMBAYRZCbjC}IIpj\`",
        "i": 0,
      },
      Object {
        "f": "eMBAYRZCbjC}IIpj\`",
        "i": 1,
      },
      Object {
        "f": "eMBAYRZCbjC}IIpj\`",
        "i": 2,
      },
    ],
    "i": 0,
    "isAromatic": false,
    "kind": 1,
    "order": 1,
    "ringSize": 0,
    "selected": true,
  },
  Object {
    "atom1": 1,
    "atom2": 2,
    "fragments": Array [
      Object {
        "atomMap": Array [],
        "idCode": "eMBAYRZ@",
        "mf": "C2H5",
      },
      Object {
        "atomMap": Array [],
        "idCode": "eFBAYc@@",
        "mf": "CH3",
      },
    ],
    "hoses1": Array [
      Object {
        "f": "eMBAYRZCbjC}IIpj\`",
        "i": 0,
      },
      Object {
        "f": "eMBAYRZCbjC}IIpj\`",
        "i": 1,
      },
      Object {
        "f": "eMBAYRZCbjC}IIpj\`",
        "i": 2,
      },
    ],
    "hoses2": Array [
      Object {
        "f": "eFBAYcAqUA~ddxUP",
        "i": 0,
      },
      Object {
        "f": "eFBAYcAqUA~ddxUP",
        "i": 1,
      },
      Object {
        "f": "eFBAYcAqUA~ddxUP",
        "i": 2,
      },
    ],
    "i": 1,
    "isAromatic": false,
    "kind": 1,
    "order": 1,
    "ringSize": 0,
    "selected": true,
  },
]
`;
